src/proxy/inbound.rs

```diff
@@ -118,13 +130,21 @@ impl Inbound {
                 .http2_initial_stream_window_size(self.cfg.window_size)
                 .http2_initial_connection_window_size(self.cfg.connection_window_size)
                 .http2_max_frame_size(self.cfg.frame_size)
-                .serve(service);
+                .serve(service)
+                .with_graceful_shutdown(async {
+                    let shutdown = self.drain.signaled().await;
+                    // TODO: add some more graceful shutdown way
+                    tokio::time::sleep(tokio::time::Duration::from_secs(3)).await;
```

Comment:
added 3s here because termination_grace_period is 5s. 

Definitely it needs otherway here